app/assets/javascripts/admin/timeline/select_all.coffee

```diff
@@ -0,0 +1,59 @@
+$ ->
+  $('#choose-all').click (e) ->
+    control_all(e)
+
+  $('#check-all-ctrl').click (e) ->
+    check_all(e)
+
+  $('#uncheck-all-ctrl').click (e) ->
+    uncheck_all(e)
+
+  $('#check-completed-ctrl').click (e) ->
+    uncheck_all(e)
+    check_completed(e)
+
+  $('#check-incomplete-ctrl').click (e) ->
+    uncheck_all(e)
+    check_incomplete(e)
+
+  # Check or uncheck all items with master checkbox
+  control_all = (e) ->
+    control = e.currentTarget
+    $inputs = $(control).closest(".timeline form").find('.select-step')
+
+    if ($(control)[0].checked == true)
+      check_items($inputs)
+    else
+      uncheck_items($inputs)
+
+  check_all = (e) ->
+    control = e.currentTarget
+    $inputs = $(control).closest(".timeline form").find('.select-step')
+    check_items($inputs)
+
+    $master_checkbox = $(control).closest(".btn-group").find("#choose-all")
+    check_items($master_checkbox)
+
+  uncheck_all = (e) ->
+    control = e.currentTarget
+    $inputs = $(control).closest(".timeline form").find('.select-step')
+    uncheck_items($inputs)
+
+    $master_checkbox = $(control).closest(".btn-group").find("#choose-all")
```

Comment:
I think that's probably overkill for now, if it were shared more places it might be worth creating a separate function for that, but this is pretty straightforward.